File: solutions/lua/pythagorean-triplet/1/pythagorean-triplet.lua
return function(sum)
  local res = setmetatable({}, { __index = table })
  for a = 3, sum // 3 do
    local b_plus_c = sum - a
    local b = (b_plus_c * b_plus_c - a * a) // (2 * b_plus_c)
    local c = sum - a - b
    if a < b and a * a + b * b == c * c then res:insert({ a, b, c }) end
  end
  return res
end

File: solutions/ruby/pythagorean-triplet/1/pythagorean_triplet.rb
module PythagoreanTriplet
  def self.triplets_with_sum(sum)
    (3..sum / 3).filter_map do |a|
      b_plus_c = sum - a
      b = (b_plus_c * b_plus_c - a * a) / (2 * b_plus_c)
      c = sum - a - b
      [a, b, c] if a < b && a * a + b * b == c * c
    end
  end
end
